api/services/User/index.js

```diff
@@ -38,48 +38,88 @@ export default class User extends BaseModelService {
     }
   }
 
+  static doesKeyBelongToInvite(key, invite) {
+    if (key.user_id !== invite.user_id) {
+      return false;
+    }
+
+    const { excludedEvents = [] } = key;
+    if (excludedEvents.includes(invite.event.id)) {
+      return false;
+    }
+
+    if (
+      invite.event.start_at >= key.dateTimeRange.start_at &&
+      invite.event.start_at <= key.dateTimeRange.end_at
+    ) {
+      return true;
+    }
+
+    if (
+      invite.event.end_at >= key.dateTimeRange.start_at &&
+      invite.event.end_at <= key.dateTimeRange.end_at
+    ) {
+      return true;
+    }
+
+    if (
+      invite.event.start_at <= key.dateTimeRange.start_at &&
+      invite.event.end_at >= key.dateTimeRange.end_at
+    ) {
+      return true;
+    }
```

Comment:
This logic is same as from `overlapDateTimeClause`

Reply:
can't you use the deep equal lib.

Reply:
How will `deep-equal` be used here. Isn't it used to compare if two objects are equal ? Here we want to check if key timings are falling between event timings